update Using the dbt IDE article (#1792)
fishtown-analytics/docs.getdbt.com: website/docs/docs/running-a-dbt-project/using-the-dbt-ide.md

id: "using-the-dbt-ide"
---

The dbt Integrated Development Environment (IDE) is a single interface for building, testing, running, and version controlling dbt projects.

### Requirements

> - The dbt IDE is powered by the [dbt-rpc](rpc) which has been overhauled in dbt v0.15.0. In order to use the IDE, your dbt project must be compatible with dbt v0.15.0.

Reviewer: ```suggestion
- The dbt IDE is powered by [dbt-rpc](rpc) which has been overhauled in dbt v0.15.0. In order to use the IDE, your dbt project must be compatible with dbt v0.15.0.
```

> To create a Development Environment, navigate to the Environments page for your Project and click the "New Environment" button.

Reviewer: ```suggestion
To create a development environment, choose **Deploy** > **Environments** from the top left. Then, click **Create Environment**.
```

Reviewer: question: is there a link we can use to get to this Environments page? it would greatly improve these navigation steps. right now, i can only get to it from the IDE

Reviewer 2: @nghi-ly I don't see a way to link there directly. How is this?
```suggestion
To create a development environment, click **Deploy**, select **Environments**, then click **Create Environment**.
```

> <Lightbox src="/img/docs/running-a-dbt-project/using-the-dbt-ide/empty-env-page.png" title="Creating a new environment for the Analytics project"/>

Reviewer: there's a mismatch here between the screenshot and the paragraph. looks like this screenshot is of the Environments page under the Deploy menu (which has the **Create** button). and the paragraph is referring to this Environments page from the IDE (which has the **New** button and i believe is the one we want): 

![Screen Shot 2022-07-27 at 12 38 49 PM](https://user-images.githubusercontent.com/107218380/181357978-9eba3ff5-e9ee-4f2f-8050-5ec519bc5fc5.png)

Author: You're right. I'm referring to the Environments page under "Deploy". Your screenshot is from the old UI

I think the text is wrong. It should say "Create Environment"

![Screen Shot 2022-07-27 at 2 31 06 PM](https://user-images.githubusercontent.com/39397591/181376067-89fa1ecf-e9e9-49ac-90fe-f894ea109c59.png)

Reviewer: i'm still unsure. the steps in this section after clicking the button talk about changing the environment type field to **Development** but that's not possible from "Create environment" (it's stuck on **Deployment** for me): 

![Screen Shot 2022-07-27 at 2 43 57 PM](https://user-images.githubusercontent.com/107218380/181379084-3146d44d-850b-486c-b1ff-ec1cdf814fcd.png)

![Screen Shot 2022-07-27 at 2 53 02 PM](https://user-images.githubusercontent.com/107218380/181379265-95a4a19d-69a8-4c54-9d4e-dd7ddc9ab88e.png)

is my dbt Cloud account not using the latest from the UI refresh; is that why it's stuck on **Deployment**, @bossnunta ?

Author: You're in the right page --- you can only have one Development environment. But you can have multiple Deployment environment. So if you already an existing Development environement, then you can't add more.

Reviewer: oh i see! thanks for the clarification, @bossnunta !

> To create a Development Environment, change the environment `Environment Type` to `Development`. Enter a name like "Dev" for the environment to help identify it amongst your other environments. During this step, you can also use a dbt Version. We recommend you selecting the same dbt Version that you're planning to use in `Deployment` environment. Finally, click "Save" to finish creating your Development Environment.

Reviewer: ```suggestion
Enter an environment **Name** that would help you identify it among your other environments (for example, `Nate's Development Environment`). Choose **Development** as the **Environment Type**. You can also select which **dbt Version** to use at this time. For compatibility reasons, we recommend that you select the same dbt version that you plan to use in your deployment environment. Finally, click **Save** to finish creating your development environment.
```

Reviewer: suggs: 

- The clause "To create a Development Environment" was used twice (back-to-back paragraphs) so reworded the paragraph a bit to remove the redundancy. 
- Applied font decorations (bold, code font) to UI fields and buttons
- Updated example name in the paragraph to match what's used in the screenshot

Reviewer: question: why should the user use the same dbt version? do they encounter errors, compatibility issues, or something else? 

would love to add that extra bit of "why you should do this" reason in the paragraph.

<Lightbox src="/img/docs/running-a-dbt-project/using-the-dbt-ide/create-dev-env.png" title="Creating a development environment"/>

### Setting up developer credentials

The IDE uses *developer credentials* to connect to your database. These developer credentials should be specific to your user. They should *not* be super user credentials, or the same credentials that you use for your production deployment of dbt.

New dbt Cloud accounts should have developer credentials created automatically as a part of Project creation in the initial application setup.

<Lightbox src="/img/docs/running-a-dbt-project/using-the-dbt-ide/dev-cred-project-setup.png" title="Developer credentials are created during project setup"/>

> New users on existing accounts *may not* have development credentials configured already. To manage your development credentials, navigate to your `Settings` in dbt Cloud and under `Your Profile` section, you can find the relevant Project under the **Credentials** header. After entering your developer credentials, you'll be able to access the dbt IDE.

Reviewer: ```suggestion
New users on existing accounts *might not* have their development credentials already configured. To manage your development credentials, go to the [Credentials](https://cloud.getdbt.com/next/settings/profile#credentials) section. Select the relevant project in the list. After entering your developer credentials, you'll be able to access the dbt IDE.
```

Reviewer: found a link to the Credentials section so sugg using that to make the navigation steps a lot more simple
